src/app/services/xml-parsers/named-entities-parser.service.ts

```diff
@@ -0,0 +1,343 @@
+import { Injectable } from '@angular/core';
+import { AttributesMap } from 'ng-dynamic-component';
+import { combineLatest } from 'rxjs';
+import { map, shareReplay } from 'rxjs/operators';
+import { AppConfig } from '../../app.config';
+import { GenericElementComponent } from '../../components/generic-element/generic-element.component';
+import {
+  Description, NamedEntitiesList, NamedEntity, NamedEntityInfo,
+  NamedEntityLabel, NamedEntityType, Relation, XMLElement,
+} from '../../models/evt-models';
+import { isNestedInElem, xpath } from '../../utils/dom-utils';
+import { EditionDataService } from '../edition-data.service';
+import { GenericParserService } from './generic-parser.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class NamedEntitiesParserService {
+  public readonly parsedLists$ = this.editionDataService.parsedEditionSource$.pipe(
+    map((source) => this.parseLists(source)),
+    shareReplay(1),
+  );
+
+  public readonly persons$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => (this.getResultsByType(lists, entities, ['person', 'personGrp']))),
+  );
+
+  public readonly places$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => this.getResultsByType(lists, entities, ['place'])),
+  );
+
+  public readonly organizations$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => this.getResultsByType(lists, entities, ['org'])),
+  );
+
+  public readonly relations$ = this.parsedLists$.pipe(
+    map(({ relations }) => relations),
+  );
+
+  public readonly events$ = this.parsedLists$.pipe(
+    map(({ lists, entities }) => this.getResultsByType(lists, entities, ['event'])),
+  );
+
+  public readonly namedEntities$ = combineLatest([
+    this.persons$,
+    this.places$,
+    this.organizations$,
+    this.relations$,
+    this.events$,
+  ]).pipe(
+    map(ne => ({
+      persons: ne[0],
+      places: ne[1],
+      organizations: ne[2],
+      relations: ne[3],
+      events: ne[4],
+    })),
+    shareReplay(1),
+  );
+
+  private neListsConfig = AppConfig.evtSettings.edition.namedEntitiesLists || {};
+  private tagNamesMap: { [key: string]: string } = {
+    persons: 'listPerson',
+    places: 'listPlace',
+    orgs: 'listOrg',
+    relations: 'listRelation',
+    events: 'listEvent',
+  };
+
+  constructor(
+    private editionDataService: EditionDataService,
+    private genericParserService: GenericParserService,
+  ) {
+  }
+
+  private parseLists(document: XMLElement) {
+    const listsToParse = this.getListsToParseTagNames();
+    const parsedLists: NamedEntitiesList[] = [];
+    let parsedEntities: NamedEntity[] = [];
+    let parsedRelations: Relation[] = [];
+    if (listsToParse) {
```

Comment:
no need of this check, `getListsToParseTagNames` always returns an array (maybe empty but not undefined or null)